src/fundus/scraping/session.py

```diff
@@ -0,0 +1,93 @@
+from typing import Optional
+
+import requests.adapters
+
+from fundus.logging import basic_logger
+
+_default_header = {"user-agent": "Fundus"}
+
+
+class SessionHandler:
+    """Object for handling  project global request.Session
+
+    The session life cycle consists of three steps which can be repeated indefinitely:
+    Build, Supply, Teardown.
+    Initially there is no session build within the session handler. When a session is requested
+    with get_session() either a new one is created with _session_factory() or the session handler's
+    existing one returned. Every subsequent call to get_session() will return the same
+    response.Session object. If close_current_session() is called, the current session will be
+    tear-downed and the next call to get_session() will build a new session.
+    """
+
+    def __init__(self, pool_connections: int = 50, pool_maxsize: int = 50):
+        self.session: Optional[requests.Session] = None
+        self.pool_connections = pool_connections
+        self.pool_maxsize = pool_maxsize
+
+    def _session_factory(self) -> requests.Session:
+        """Builds a new Session
+
+        This returns a new client session build from pre-defined configurations:
+        - pool_connections: 50
+        - pool_maxsize: 50
+        - hooks = {'response': raise_for_status(), _response_log():}
```

Comment:
The docstring for the hooks look malformed, but I'm not really sure what should go here.